d3a_interface/sim_results/bills.py

```diff
@@ -348,32 +377,59 @@ def _generate_external_and_total_bills(self, area_dict, results):
+            totals_child_list = [results[area_dict['uuid']]["Accumulated Trades"],
+                                 results[area_dict['uuid']]["External Trades"],
+                                 results[area_dict['uuid']]["Market Fees"]]
         else:
             # If root area, Accumulated Trades == Totals
-            results[area_dict['name']].update(self._market_fee_section(total_market_fee))
-            totals_child_list = [results[area_dict['name']]["Accumulated Trades"],
-                                 results[area_dict['name']]["Market Fees"]]
+            results[area_dict['uuid']].update(self._market_fee_section(total_market_fee))
+            totals_child_list = [results[area_dict['uuid']]["Accumulated Trades"],
+                                 results[area_dict['uuid']]["Market Fees"]]
 
         self._write_accumulated_stats(area_dict, results, totals_child_list, "Totals")
         return results
 
-    def _bills_for_redis(self, area_dict, bills_results):
-        if area_dict['name'] in bills_results:
-            self.bills_redis_results[area_dict['uuid']] = \
-                self._round_area_bill_result_redis(bills_results[area_dict['name']])
+    @classmethod
+    def _bills_local_format(self, area_dict, bills_results_uuids):
+        bills_results = {}
+        if area_dict['uuid'] in bills_results_uuids:
+            bills_results[area_dict['name']] = bills_results_uuids[area_dict['uuid']]
         for child in area_dict['children']:
             if child['children']:
-                self._bills_for_redis(child, bills_results)
-            elif child['name'] in bills_results:
-                self.bills_redis_results[child['uuid']] = \
-                    self._round_child_bill_results(bills_results[child['name']])
+                bills_results.update(self._bills_local_format(child, bills_results_uuids))
+            elif child['uuid'] in bills_results_uuids:
+                bills_results[child['name']] = bills_results_uuids[child['uuid']]
+        return bills_results
+
+    @classmethod
+    def _create_uuid_name_mapping(cls, area_dict, results):
+        results[area_dict["uuid"]] = area_dict["name"]
+        if key_in_dict_and_not_none(area_dict, "children"):
+            for child in area_dict["children"]:
+                results.update(cls._create_uuid_name_mapping(child, results))
+        return results
+
+    def _swap_children_uuids_to_names(self, area_dict, bills_results):
+        area_uuid_name_mapping = self._create_uuid_name_mapping(area_dict, {})
+        final_result = {}
+        for k, children in bills_results.items():
+            children_result = {}
+            if "bought" in children:
+                final_result[k] = children
+                continue
+            for uuid, child in children.items():
+                if uuid in ["Totals", "Accumulated Trades", "External Trades", "Market Fees"] or \
+                        uuid not in area_uuid_name_mapping:
```

Comment:
This surprised my t first, but it is really only a precaution, right?

Reply:
I assume you are talking about the `uuid not in area_uuid_name_mapping`, yes it is a precaution in order to not lose entries by mistake. Only the `uuid in ["Totals", "Accumulated Trades", "External Trades", "Market Fees"]` should be sufficient